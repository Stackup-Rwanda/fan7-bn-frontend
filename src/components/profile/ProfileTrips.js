import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import moment from 'moment';
import './styles/profileTrips.scss';

const ProfileTrips = ({ trips }) => (
  <div className="trips">
    <h1 className="trips__title">Recent Trip Requests</h1>
    <table className="trips__table">
      <thead>
        <tr>
          <td>Destination</td>
          <td> Travel Date</td>
          <td> Status</td>
        </tr>
      </thead>
      <tbody>
        {trips
          ? trips.map((trip) => (
            <tr>
              <td>{trip.destination.map((destination) => destination)}</td>
              <td>{trip.travel_date.map((travelDate) => moment(travelDate).format('L'))}</td>
              <td>{trip.status}</td>
            </tr>
          ))
          : (
            <tr>
              <td colSpan="3">No trip requests found</td>
            </tr>
          )}
      </tbody>
    </table>
    <Link to="/trip-requests" className="trips__more">
      View more
    </Link>
  </div>
);

ProfileTrips.propTypes = {
  trips: PropTypes.arrayOf().isRequired,
};

export default ProfileTrips;
