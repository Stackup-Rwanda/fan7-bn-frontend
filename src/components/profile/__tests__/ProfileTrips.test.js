/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import ProfileTrips from '../ProfileTrips';

const history = {
  push: jest.fn(),
  replace: jest.fn(),
  createHref: jest.fn(),
};

describe('ProfileTrips Component Test', () => {
  it('should return true if component exists', () => {
    const wrapper = mount(
      <Router history={history}>
        <ProfileTrips />
      </Router>,
    );

    expect(wrapper.find('.trips').exists()).toBe(true);
  });

  it('should display appropriate props', () => {
    const props = {
      trips: [{ destination: ['Nigeria, Lagos'], travel_date: ['2020-5-6'], status: 'pending' }],
    };
    const wrapper = shallow(<ProfileTrips {...props} />);

    expect(
      wrapper
        .find('tbody tr td')
        .first()
        .text(),
    ).toEqual('Nigeria, Lagos');
  });

  it('should display no trip  requests found', () => {
    const props = {
      trips: null,
    };
    const wrapper = shallow(<ProfileTrips {...props} />);

    expect(
      wrapper
        .find('tbody tr td')
        .first()
        .text(),
    ).toEqual('No trip requests found');
  });
});

describe('<ProfileTrips /> shallow rendering tests', () => {
  it('matches the snapshot', () => {
    const props = {
      trips: [{ destination: ['Nigeria, Lagos'], travel_date: ['2020-5-6'], status: 'pending' }],
    };
    const tree = shallow(<ProfileTrips {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
