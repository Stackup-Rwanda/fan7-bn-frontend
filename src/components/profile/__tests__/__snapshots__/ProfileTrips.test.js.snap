// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileTrips /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="trips"
>
  <h1
    className="trips__title"
  >
    Recent Trip Requests
  </h1>
  <table
    className="trips__table"
  >
    <thead>
      <tr>
        <td>
          Destination
        </td>
        <td>
           Travel Date
        </td>
        <td>
           Status
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Nigeria, Lagos
        </td>
        <td>
          05/06/2020
        </td>
        <td>
          pending
        </td>
      </tr>
    </tbody>
  </table>
  <Link
    className="trips__more"
    to="/trip-requests"
  >
    View more
  </Link>
</div>
`;
